File: feature-map/src/main/java/home/oleg/placesnearme/feature_map/custom/LoadingView.java
package home.oleg.placesnearme.feature_map.custom;

import android.content.Context;
import android.support.annotation.Nullable;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.widget.Button;
import android.widget.FrameLayout;
import android.widget.LinearLayout;

import com.github.ybq.android.spinkit.SpinKitView;

import butterknife.BindView;
import butterknife.ButterKnife;
import home.oleg.placesnearme.feature_map.R;
import home.oleg.placesnearme.feature_map.R2;

public class LoadingView extends FrameLayout {

    @BindView(R2.id.spinKit)
    SpinKitView spinKitView;

    @BindView(R2.id.retryButton)
    Button retryButton;

    public LoadingView(Context context) {
        this(context, null);
    }

    public LoadingView(Context context, @Nullable AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public LoadingView(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        this(context, attrs, defStyleAttr, 0);
    }

    public LoadingView(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
        init();
    }

    private void init() {
        LayoutInflater.from(getContext()).inflate(R.layout.view_loading, this, true);
        ButterKnife.bind(this);
    }

    public void showLoading() {
        this.setVisibility(VISIBLE);
        retryButton.setVisibility(GONE);
        spinKitView.setVisibility(VISIBLE);
    }

    public void hideLoading() {
        this.setVisibility(GONE);
        retryButton.setVisibility(GONE);
        spinKitView.setVisibility(GONE);
    }

    public void showError() {
        this.setVisibility(VISIBLE);
        retryButton.setVisibility(VISIBLE);
        spinKitView.setVisibility(GONE);
    }
}

File: feature-map/src/main/java/home/oleg/placesnearme/feature_map/presentation/ui/LoadingView.kt
package home.oleg.placesnearme.feature_map.presentation.ui

import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import android.view.View
import android.widget.FrameLayout

import home.oleg.placesnearme.feature_map.R
import kotlinx.android.synthetic.main.view_loading.view.*

class LoadingView @JvmOverloads constructor(
        context: Context, attrs: AttributeSet? = null,
        defStyleAttr: Int = 0,
        defStyleRes: Int = 0) : FrameLayout(context, attrs, defStyleAttr, defStyleRes) {

    init {
        init()
    }

    private fun init() {
        LayoutInflater.from(context).inflate(R.layout.view_loading, this, true)
    }

    fun setOnRetryCLickListener(onRetryCLickListener: () -> Unit) {
        retryButton.setOnClickListener {
            onRetryCLickListener()
        }
    }

    fun showLoading() {
        this.visibility = View.VISIBLE
        retryButton.visibility = View.GONE
        spinKit.visibility = View.VISIBLE
    }

    fun hide() {
        this.visibility = View.GONE
        retryButton.visibility = View.GONE
        spinKit.visibility = View.GONE
    }

    fun showRetry() {
        this.visibility = View.VISIBLE
        retryButton.visibility = View.VISIBLE
        spinKit.visibility = View.GONE
    }
}
